Cargo.toml

```diff
@@ -92,6 +92,7 @@ derivative = { version = "2" }
 derive_more = { version = "0.99" }
 enum-iterator = "1.2"
 hyper = { version = "0.14.26" }
+num = "0.4.3"
```

Comment:
Could you use directly [num-rational](https://github.com/rust-num/num-rational) please?=)

Reply:
Done